carma-messenger-core/cpp_message/src/Map_Message.cpp

```diff
@@ -0,0 +1,614 @@
+                    {
+                        new_intersection.name_exists=true;
+                        for (size_t n = 0 ; n < map_msg_intersections->name->size; n++)
+                        {
+                            new_intersection.name.push_back(n);
+                        }
+                    }
+                    else
+                    {
+                        new_intersection.name_exists = false;
+                    }
+
+                    if(map_msg_intersections->laneWidth)
+                    {
+                        new_intersection.lane_width_exists = true;
+                        auto width = map_msg_intersections->laneWidth;
+                        new_intersection.lane_width = *width;
+                    }
+                    else
+                    {
+                        new_intersection.lane_width_exists = false;
+                    }        
+
+                    if(map_msg_intersections->preemptPriorityData)
+                    {
+                        new_intersection.preempt_priority_data_exists = true;
+
+                        for(size_t p = 0; i<map_msg_intersections->preemptPriorityData->list.size; p++)
+                        {
+                            SignalControlZone_t *sig = new SignalControlZone_t;
+                            sig = map_msg_intersections->preemptPriorityData->list.array[p];
+                            
+                            j2735_msgs::SignalControlZone s;
+
+                            /*SignalControlZone.msg states that RegionalExtension has not yet been implemented*/
+
+                            new_intersection.preempt_priority_data.preempt_priority_list.push_back(s);
+                        }
+                        
+                    }
+                    else
+                    {
+                        new_intersection.preempt_priority_data_exists = false;
+                    }
+
+                    if(map_msg_intersections->speedLimits)
+                    {
+                        new_intersection.speed_limits_exists = true;
+                        
+                        for(size_t s = 0; s < map_msg_intersections->speedLimits->list.size ; s++)
+                        {
+                            RegulatorySpeedLimit_t *speedLimit = new RegulatorySpeedLimit_t;
+
+                            speedLimit = map_msg_intersections->speedLimits->list.array[s];
+
+                            j2735_msgs::RegulatorySpeedLimit rsl;
+                            rsl.speed = speedLimit->speed;
+                            rsl.type.speed_limit_type = speedLimit->type;
+                            new_intersection.speed_limits.speed_limits.push_back(rsl);
+                        }
+                    }
+                    else
+                    {
+                        new_intersection.speed_limits_exists = false;
+                    }
+
+                    if(map_msg_intersections->refPoint.elevation) 
```

Comment:
The refPoint.elevation and the associated lat/lon values were already structured this way in the header and .msg files.

Reply:
Do you mean its defined this way for the min/max values? we should still move the lat/lon outside the bool for elevation. There is a chance that elevation is not defined since it is optional, lat/lon shouldn't be ignored in that case.